### PR TITLE
Add the WSO2 vs. Ballerina case study

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the WSO2 vs. Ballerina case study.
> Fixes #

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -387,6 +387,10 @@ const nextConfig = {
       {
         source: `/learn/pre-built-integrations/sample2`,
         destination: `/learn/integration/pre-built-integrations/sample2`,
+      },
+      {
+        source: `/learn/case-studies/wso2-vs-ballerina`,
```
Comment (sm1990) [inline next.config.js:392]: Remove this block. Case studies does not come under Learn. The url should be b.io/case-studies

File: pages/case-studies/[slug].js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import Head from "next/head";
+
+import Layout from "../../layouts/LayoutOther";
+import MainContent from "../../components/common/main-content/MainContent";
+import { prefix } from "../../utils/prefix";
+import Toc from "../../components/common/pg-toc/Toc";
+
+export async function getStaticPaths() {
+    // Retrieve all our slugs
+    const files = fs.readdirSync('policy');
```
Comment (sm1990) [inline pages/case-studies/[slug].js:33]:
```suggestion
    const files = fs.readdirSync('case-studies');
```

File: pages/case-studies/[slug].js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    return {
+        paths,
+        fallback: false,
+    };
+}
+
+export async function getStaticProps({ params: { slug } }) {
+
+    const fileName = fs.readFileSync(`policy/${slug}.md`, 'utf-8');
```
Comment (sm1990) [inline pages/case-studies/[slug].js:48]:
```suggestion
    const fileName = fs.readFileSync(`case-studies/${slug}.md`, 'utf-8');
```

File: pages/case-studies/[slug].js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            </Head>
+            <Layout>
+                <Col xs={12} sm={10} className="policyContent">
+                    <Row className="topRow innerRow">
+                        <Col xs={11}>
+                            <h1>{frontmatter.title}</h1>
+                        </Col>
+                        <Col xs={1} className="gitIcon">
+                            <a
+                                href={`${process.env.gitHubPath}policy/${slug}.md`}
```
Comment (sm1990) [inline pages/case-studies/[slug].js:116]:
```suggestion
                                href={`${process.env.gitHubPath}case-studies/${slug}.md`}
```

File: swan-lake/case-studies/wso2-vs-ballerina.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+---
```
Comment (sm1990) [inline swan-lake/case-studies/wso2-vs-ballerina.md:1]: Move the case-studies directory to the root level

Comment (praneesha) [reply swan-lake/case-studies/wso2-vs-ballerina.md]: Fixed.

File: next.config.js (hunk 1 above)
Comment (praneesha) [reply next.config.js:392]: Removed.

**Review by sm1990 — APPROVED** [review]

**Pull request merged**